libgalois/src/PropertyGraph.cpp

```diff
@@ -1289,3 +1306,16 @@ katana::PropertyGraph::GetNodePropertyIndex(
   }
   return KATANA_ERROR(katana::ErrorCode::NotFound, "node index not found");
 }
+
+katana::Result<void>
+katana::PropertyGraph::RecreatePropertyIndexes() {
+  for (const std::string& column_name : rdg_.node_property_index_columns()) {
+    KATANA_CHECKED(MakeNodeIndex(column_name));
+  }
+
+  for (const std::string& column_name : rdg_.edge_property_index_columns()) {
+    KATANA_CHECKED(MakeEdgeIndex(column_name));
+  }
+
+  return katana::ResultSuccess();
+}
```

Comment:
I encountered some issues when i was testing the code. For instance, a column of type `int32_t` would cause the indexing code to crash - which we don't want. But there might be cases where we would like it to crash - for instance specifying an index for a column that doesn't exist. I don't have any concrete suggestions for here - probably some points to think about.

Reply:
Crash or return an error? If it's crashing then that's bad, and I'd like to fix that. But returning an error here seems like the right thing to do when asking to index a column for an as-yet-unsupported data type, or a column that does not exist.